### PR TITLE
configure: only req CMake if we're building LLVM

### PR DESCRIPTION
**Pull request by cardoe**
CMake is only necessary if LLVM is going to be built and not in any
other case.

Signed-off-by: Doug Goldstein <cardoe@cardoe.com>

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -848,7 +848,10 @@ then
 fi
 
 # For building LLVM
-probe_need CFG_CMAKE cmake
+if [ -z $CFG_LLVM_ROOT ]
```
Comment (jmesmon) [inline configure]: Should this be quoted? `if [ -z "$CFG_LLVM_ROOT"]`?

**Comment by alexcrichton** [comment]
Looks good to me, but I agree that we probably need quotes (or at least I think we do...)

**Comment by cardoe** [comment]
Fixed. I agree it should be quoted. I was actually copying the behavior in other places in the script.
```
    elif [ -z $CFG_LLVM_ROOT ]
```

I'll add another commit that quotes that.

**Pull request merged**